launchable/test_runners/playwright.py

```diff
@@ -49,3 +61,184 @@ def subset(client):
+                                                "retry": 0,
+                                                "startTime": "2024-04-08T07:42:26.319Z",
+                                                "attachments": []
+                                            }
+                                        ],
+                                        "status": "expected"
+                                    }
+                                ],
+                                "id": "f8b37aaddd8ecd14ecd4-3367671d4d7af1046962",
+                                "file": "tests/demo-todo-app.spec.ts",
+                                "line": 14,
+                                "column": 7
+                            }
+                        ]
+                    }
+                ]
+            }
+        ]
+    }
+    """
+
+    def __init__(self, client):
+        self.client = client
+
+    def parse_func(self, report_file: str) -> Generator[CaseEvent, None, None]:
+        data: Dict[str, Dict]
+        with open(report_file, 'r') as json_file:
+            try:
+                data = json.load(json_file)
+            except Exception as e:
+                raise Exception("Can't read JSON format report file {}. Make sure to confirm report file.".format(
+                    report_file)) from e
+
+        suites: List[Dict[str, Dict]] = list(data.get("suites", []))
+        if len(suites) == 0:
+            click.echo("Can't find test results from {}. Make sure to confirm report file.".format(
+                report_file), err=True)
+
+        for s in suites:
+            # The title of the root suite object contains the file name.
+            test_file = str(s.get("title", ""))
+
+            for event in self._parse_suites(test_file, s, []):
+                yield event
+
+    def _parse_suites(self, test_file: str, suite: Dict[str, Dict], test_case_names: List[str] = []) -> List:
+        events = []
+
+        # In some cases, suites are nested.
+        for s in suite.get("suites", []):
+            for e in self._parse_suites(test_file, s, test_case_names + [s.get("title")]):
+                events.append(e)
+
+        for spec in suite.get("specs", []):
+            spec_name = spec.get("title", "")
+            line_no = spec.get("line", None)
+            for test in spec.get("tests", []):
+                for result in test.get("results", []):
+                    test_path: TestPath = [
+                        self.client.make_file_path_component(test_file),
+                        {"type": "testcase", "name": TEST_CASE_DELIMITER.join(test_case_names + [spec_name])}
+                    ]
+
+                    duration_msec = result.get("duration", 0)
+                    status = self._case_event_status_from_str(result.get("status", ""))
+                    stdout = self._parse_stdout(result.get("stdout", []))
+                    stderr = self._parse_stderr(result.get("errors", []))
```

Comment:
Can we assign an array to stdout and stderr?
Optional[str] is defined as a type in these variables.
https://github.com/launchableinc/cli/blob/main/launchable/commands/record/case_event.py#L152

Reply:
Please see `_parse_xxx` methods. These methods convert from array to string. 
Does it make sense?